RecoEgamma/EgammaHLTProducers/plugins/EgammaHLTClusterShapeProducer.cc

```diff
@@ -88,45 +91,81 @@ void EgammaHLTClusterShapeProducer::produce(edm::StreamID sid,
   EcalClusterLazyTools lazyTools(iEvent, ecalClusterLazyToolsESData, ecalRechitEBToken_, ecalRechitEEToken_);
   noZS::EcalClusterLazyTools lazyTools5x5(iEvent, ecalClusterLazyToolsESData, ecalRechitEBToken_, ecalRechitEEToken_);
 
+  //Map is for sigmaIEtaIEta
   reco::RecoEcalCandidateIsolationMap clshMap(recoecalcandHandle);
   reco::RecoEcalCandidateIsolationMap clsh5x5Map(recoecalcandHandle);
   reco::RecoEcalCandidateIsolationMap clsh5x5NoiseCleanedMap(recoecalcandHandle);
 
+  //Map2 is for sigmaIPhiIPhi
+  reco::RecoEcalCandidateIsolationMap clshMap2(recoecalcandHandle);
+  reco::RecoEcalCandidateIsolationMap clsh5x5Map2(recoecalcandHandle);
+  reco::RecoEcalCandidateIsolationMap clsh5x5NoiseCleanedMap2(recoecalcandHandle);
+
   for (unsigned int iRecoEcalCand = 0; iRecoEcalCand < recoecalcandHandle->size(); iRecoEcalCand++) {
     reco::RecoEcalCandidateRef recoecalcandref(recoecalcandHandle, iRecoEcalCand);
     if (recoecalcandref->superCluster()->seed()->seed().det() != DetId::Ecal) {  //HGCAL, skip for now
       clshMap.insert(recoecalcandref, 0);
       clsh5x5Map.insert(recoecalcandref, 0);
+      clsh5x5NoiseCleanedMap.insert(recoecalcandref, 0);
```

Comment:
Just for my own understanding. This looks like a fix to fill one of the existing maps for candidates in HGCal, but I assume it has no impact on previous EGM results for Phase-2. Correct? And if so, why?

Reply:
yes this has no impact on Phase2 HLT TDR results of EGM because NoiseCleaned shower-shapes entered CMSSW only in 11_3_X via this PR https://github.com/cms-sw/cmssw/pull/33148. The HLT TDR release, 11_1_X, do not have noise-cleaned e/gamma variables. For phase2 studies in barrel until now, we have been using the usual Full5x5 showershapes, where no noise-cleaning is applied.